packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -442,7 +448,7 @@ export abstract class SharedObjectCore<TEvent extends ISharedObjectEvents = ISha
 				local: boolean,
 				localOpMetadata: unknown,
 			) => {
-				this.process(message, local, localOpMetadata);
+				this.process(parseHandles(message, this.serializer), local, localOpMetadata);
```

Comment:
About optimizing perf impact*:

`parseHandles` does a roundtrip through string, I wonder if we should update it to use `IFluidSerializer.decode` which does its own object walk?

Also, `parseHandles` only needs to be done on `message.contents`

\* If this PR was simply moving `parseHandles` calls from DDS implementations to this one place, it could be a separate follow-up.  But I see at least one place (`sharedTreeCore.ts`) where previously it was using `decode` but now will use the (presumably slower) `parseHandles`.